openshift-knative-operator/pkg/serving/ingress.go

```diff
@@ -51,3 +62,57 @@ func defaultIngressClass(ks *operatorv1beta1.KnativeServing) string {
 	}
 	return kourierIngressClassName
 }
+
+func generateDefaultIstioNetworkPoliciesIfRequired(ks base.KComponent) ([]mf.Manifest, error) {
+	if !ks.(*operatorv1beta1.KnativeServing).Spec.Ingress.Istio.Enabled {
+		return nil, nil
+	}
+
+	if v, ok := ks.GetAnnotations()[disableGeneratingIstioNetPoliciesAnnotation]; ok {
+		b, _ := strconv.ParseBool(v)
+		if b {
+			return nil, nil
+		}
+	}
+
+	unObjs := []unstructured.Unstructured{{}, {}, {}}
+	for i, name := range []string{"webhook", "net-istio-webhook", "allow-from-openshift-monitoring-ns"} {
+		nwp := networkingv1.NetworkPolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: ks.GetNamespace(),
+				Labels: map[string]string{
+					"networking.knative.dev/ingress-provider": "istio",
+				},
+			},
+		}
+		if name != "allow-from-openshift-monitoring-ns" {
```

Comment:
maybe use `switch` to make it more clear which cases are the same?

Reply:
Can do that